### PR TITLE
MERGE: Feature/template fixes

### LINKED ISSUE #162: Implement file upload dialog
Create html link to dialog/form which handles uploading a file and sending the data to dir/{ID}/upload.

Implement in directory.html

### LINKED ISSUE #235: Template link is incorrect
This is `http://localhost:8080/search` returned for file search, but this is incorrect.

The other buttons in group template are also incorrect.

### PR DESCRIPTION
**Pull request by Thabb**
We fixed a lot of the functionality. Just the search form is missing.

Closes #162
Closes #235

### CODE REVIEW

File: src/main/java/mops/presentation/DirectoryController.java [hunk 1]
```diff
@@ -143,7 +144,7 @@ public String createSubFolder(RedirectAttributes redirectAttributes,
      * @param dirId              id of the folder to be deleted
      * @return the id of the parent folder
      */
-    @DeleteMapping("/{dirId}")
+    @PostMapping("/{dirId}")
```
Comment (iTitus) [inline src/main/java/mops/presentation/DirectoryController.java]: can you make this /{dirId}/delete so it's clear what happens?

File: src/main/resources/templates/directory.html [hunk 2]
```diff
@@ -79,38 +79,41 @@
             <th>Typ</th>
             <th>Größe</th>
         </tr>
-        <tr th:each="dir : ${dirs}" th:href="@{/${dir.id}}" }>
-            <td th:text="${dir.name}">%Name%</td>
+        <tr th:each="dir : ${dirs}" }>
+            <td> <a th:text="${dir.name}" th:href="@{${dir.id}}">%Name%</a></td>
```
Comment (iTitus) [inline src/main/resources/templates/directory.html]:
```suggestion
            <td><a th:text="${dir.name}" th:href="@{/material1/dir/{dirId}(dirId=${dir.id})}">%Name%</a></td>
```

**Comment by Segelzwerg** [comment]
Can you add the issues that will be closed. Code review will follow.

File: src/main/resources/templates/directory.html [hunk 3]
```diff
@@ -79,38 +79,41 @@
...
             <td></td>
             <td>Dateiordner</td>
             <td></td>
             <td>
-                <form action="#" th:action="@{/${dir.id}}" th:method="delete">
+                <form action="#" th:action="@{${dir.id}}" th:method="post">
```
Comment (iTitus) [inline src/main/resources/templates/directory.html]:
```suggestion
                <form action="#" th:action="@{/material1/dir/{dirId}/delete(dirId=${dir.id})}" th:method="post">
```

File: src/main/resources/templates/directory.html [hunk 4]
```diff
@@ -79,38 +79,41 @@
...
                     <input type="submit" name="deleteFolder" value="Ordner löschen">
                 </form>
             </td>
         </tr>
-        <tr th:each="file : ${files}" th:href="@{../file/${file.id}}">
-            <td th:text="${file.name}">%Name%</td>
+        <tr th:each="file : ${files}">
+            <td> <a th:text="${file.name}" th:href="@{'../file/' +  ${file.id}}">%Name%</a></td>
```
Comment (iTitus) [inline src/main/resources/templates/directory.html]:
```suggestion
            <td><a th:text="${file.name}" th:href="@{/material1/file/{fileId}(fileId=${file.id})}">%Name%</a></td>
```

File: src/main/resources/templates/directory.html [hunk 5]
```diff
@@ -79,38 +79,41 @@
...
             <td th:text="TBD">%Datum%</td>
             <td th:text="${file.type}">%Typ%</td>
             <td th:text="${file.size}">%Größe%</td>
             <td>
-                <form action="#" th:action="@{../file/${file.id}}" th:method="delete">
+                <form action="#" th:action="@{'../file/' + ${file.id}}" th:method="post">
```
Comment (iTitus) [inline src/main/resources/templates/directory.html]:
```suggestion
                <form action="#" th:action="@{/material1/file/{fileId}/delete(fileId=${file.id})}" th:method="post">
```

File: src/main/resources/templates/directory.html [hunk 6]
```diff
@@ -79,38 +79,41 @@
...
                     <input type="submit" name="deleteFile" value="Datei löschen">
                 </form>
             </td>
         </tr>
     </table>
-    <form action="#" th:action="@{/create}" th:method="post">
+
+    <form action="#" th:action="@{${thisDirId} + '/create'}" th:method="post">
```
Comment (iTitus) [inline src/main/resources/templates/directory.html]:
```suggestion
    <form action="#" th:action="@{/material1/dir/{dirId}/create(dirId=${thisDirId})}" th:method="post">
```

File: src/main/resources/templates/directory.html [hunk 7]
```diff
@@ -79,38 +79,41 @@
...
         <input type="text" name="folderName">
         <input type="submit" name="newFolder" value="Neuer Ordner">
     </form>
-    <form action="#" th:action="@{/upload}" th:method="post">
+
+    <form action="#" th:action="@{${thisDirId} + '/upload'}" th:method="post" enctype="multipart/form-data">
```
Comment (iTitus) [inline src/main/resources/templates/directory.html]:
```suggestion
    <form action="#" th:action="@{/material1/dir/{dirId}/upload(dirId=${thisDirId})}" th:method="post" enctype="multipart/form-data">
```

File: src/main/resources/templates/directory.html [hunk 8]
```diff
@@ -79,38 +79,41 @@
...
         <input type="file" name="file">
         <input type="submit" name="uploadFile" value="Datei Hochladen">
     </form>
-    <form action="#" th:action="@{/search}" th:method="post" th:object="${fileQueryForm}">
+
+    <form action="#" th:action="@{${thisDirId} + '/search'}" th:method="post" th:object="${fileQueryForm}">
```
Comment (iTitus) [inline src/main/resources/templates/directory.html]:
```suggestion
    <form action="#" th:action="@{/material1/dir/{dirId}/search(dirId=${thisDirId})}" th:method="post" th:object="${fileQueryForm}">
```

File: src/main/java/mops/presentation/FileController.java [hunk 9]
```diff
@@ -116,7 +116,7 @@
      * @param fileId             the id of the file to be deleted
      * @return the route to the parentDir of the deleted file
      */
-    @DeleteMapping("/{fileId}")
+    @PostMapping("/{fileId}")
```
Comment (Segelzwerg) [inline src/main/java/mops/presentation/FileController.java]: Probably the same as above.

**Comment by iTitus** [comment]
> 
> 
> One more delete method should be renamed.
> I'm not sure about @iTitus suggestions. @Thabb is shorter and easier to read and as long as it does work.

String concatenation is inherently more inelegant than templating.
Concatenation also has security risks.
Safer and easier to always use templating

**Comment by Segelzwerg** [comment]
> > One more delete method should be renamed.
> > I'm not sure about @iTitus suggestions. @Thabb is shorter and easier to read and as long as it does work.
> 
> String concatenation is inherently more inelegant than templating.
> Concatenation also has security risks.
> Safer and easier to always use templating

I see your point, but is there a way of not to use the whole prefix? And it is not really a concatenation as maybe a sql injection.

**Comment by Segelzwerg** [comment]
And we need to update documentation for the changed routes.

**Comment by Thabb** [comment]
Closes #162 
Closes #235

**Pull request merged**